Tools/scripts/dumpcore.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+# attaches GDB to a process to dump a core
+
+# we want everything on stderr, so the program is not disturbed
+exec 1>&2
+
+PID=$1
```

Comment:
did you try using /usr/bin/gcore ?

Reply:
I did not - 'though I looked around to find an equivalent to that command :-)

I know this works, 'though :-)

![image](https://user-images.githubusercontent.com/7077857/124581154-30315e80-de94-11eb-9e74-98642b2ad39d.png)

Core files aren't of immediate use on my machine; gdb won't load the ArduPilot executable.  But we do get them:

```
pbarker@bluebottle:/tmp/fred$ file tmp/buildlogs/ap-7725.core-QuadPlane-PIDTuning
tmp/buildlogs/ap-7725.core-QuadPlane-PIDTuning: ELF 64-bit LSB core file x86-64, version 1 (SYSV), SVR4-style, from '/__w/ardupilot/ardupilot/build/sitl/bin/arduplane', real uid: 0, effective uid: 0, real gid: 0, effective gid: 0, execfn: '/__w/ardupilot/ardupilot/build/sitl/bin/arduplane', platform: 'x86_64'
pbarker@bluebottle:/tmp/fred$ gdb tmp/buildlogs/arduplane tmp/buildlogs/ap-7725.core-QuadPlane-PIDTuning
GNU gdb (Ubuntu 8.1.1-0ubuntu1) 8.1.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
/home/pbarker/.gdbinit:8: Error in sourced command file:
/dev/serial/by-id/usb-Black_Sphere_Technologies_Black_Magic_Probe_E3B9CCB9-if00: No such file or directory.
Reading symbols from tmp/buildlogs/arduplane...done.

warning: exec file is newer than core file.
[New LWP 7711]
[New LWP 7713]

warning: .dynamic section for "/lib/x86_64-linux-gnu/libpthread.so.0" is not at the expected address (wrong library or version mismatch?)

warning: Could not load shared library symbols for /lib/x86_64-linux-gnu/libstdc++.so.6.
Do you need "set solib-search-path" or "set sysroot"?

warning: File "/lib/x86_64-linux-gnu/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".
To enable execution of this file add
	add-auto-load-safe-path /lib/x86_64-linux-gnu/libthread_db-1.0.so
line to your configuration file "/home/pbarker/.gdbinit".
To completely disable this security protection add
	set auto-load safe-path /
line to your configuration file "/home/pbarker/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
	info "(gdb)Auto-loading safe path"

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.

warning: File "/lib/x86_64-linux-gnu/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
Core was generated by `/__w/ardupilot/ardupilot/build/sitl/bin/arduplane'.
#0  0x00007f3128c8ddff in __expf128_table ()
   from /lib/x86_64-linux-gnu/libm.so.6
[Current thread is 1 (LWP 7711)]
(gdb) 
```